File: ExpressionTreeTestObjects.VB/VBCompiler/Member.vb
Partial Module VBCompiler

    <TestObject(Member)>
    Friend InstanceMember As Expression = IIFE(Function()
                                                   Dim s = ""
                                                   Return Expr(Function() s.Length)
                                               End Function)

    <TestObject(Member)>
    Friend ClosedVariable As Expression = IIFE(Function()
                                                   Dim s = ""
                                                   Return Expr(Function() s)
                                               End Function)

    <TestObject(Member)>
    Friend StaticMember As Expression = Expr(Function() String.Empty)
End Module

File: Tests/dynamicLinqTestObjects/Member.cs
using ExpressionTreeTestObjects;
using System.Linq.Expressions;
using static ExpressionTreeTestObjects.Categories;
using static ExpressionTreeTestObjects.Functions;

namespace ExpressionTreeToString.Tests {
    partial class DynamicLinqTestObjects {
        [TestObject(Member)]
        internal static readonly Expression StaticMember = Expr(p => string.Empty);

        [TestObject(Member)]
        internal static readonly Expression ParameterMember = Expr(p => p.LastName);

        [TestObject(Member)]
        internal static readonly Expression InstanceMember = Expr(p => p.LastName!.Length);

        [TestObject(Member)]
        internal static readonly Expression ClosureValue = IIFE(() => {
            var x = 5;
            return Expr(p => x);
        });
    }
}
